larq_compute_engine/mlir/transforms/optimize.cc

```diff
@@ -40,7 +39,13 @@ bool IsConstantValue(Attribute values, float expected_value) {
   return true;
 }
 
-#include "larq_compute_engine/mlir/transforms/generated_optimize.inc"
+namespace target_arm {
+#include "larq_compute_engine/mlir/transforms/generated_optimize_target_arm.inc"
+}
+
+namespace target_other {
+#include "larq_compute_engine/mlir/transforms/generated_optimize_target_other.inc"
+}
```

Comment:
These have to be included into distinct namespaces otherwise there are name conflicts.

Reply:
This approach sounds good to me. Alternatively we could not include `optimize_common.td` in `optimize_target_arm.td` and then always populate the patterns from `optimize_common.td` here and only conditionally add the target specific ones.

Reply:
The nice thing about the `include` statement is that it means we don't need to redefine the 'helper' functions like these at the top of the target-specific files:

https://github.com/larq/compute-engine/blob/95fd0b09a55647a021bed352fdacf4257c31767b/larq_compute_engine/mlir/transforms/optimize_patterns.td#L6-L10

Reply:
👍 Makes sense.